docs/self-managed/setup/deploy/amazon/amazon-eks/terraform-setup.md

```diff
@@ -250,65 +252,426 @@ access_entries = {
+https://github.com/camunda/camunda-tf-eks-module/blob/feature/opensearch-doc/examples/camunda-8.6/opensearch.tf
+```
+
+  </TabItem>
+  
+  <TabItem value="irsa" label="IRSA">
+
+In addition to standard authentication that uses anonynous user and rely on the network for the access control, you can also use [**IRSA (IAM Roles for Service Accounts)**](https://aws.amazon.com/blogs/opensource/introducing-fine-grained-iam-roles-service-accounts/) to securely connect to OpenSearch. IRSA enables your Kubernetes workloads to assume IAM roles without managing AWS credentials directly.
+
+**Note**: Using IRSA is optional. If you prefer, you can continue using password-based access to your OpenSearch domain.
+
+If you choose to use IRSA, you’ll need to take note of the **IAM role name** created for OpenSearch and the **AWS Account ID**, as these will be required later to annotate the Kubernetes service account.
+
+##### OpenSearch IRSA role and policy
+
+To configure IRSA for OpenSearch, the OpenSearch module uses outputs from the EKS cluster module to define the necessary IAM role and policies.
+
+Here's an example of how to define the IAM role trust policy and access policy for OpenSearch, this configuration will deploy an OpenSearch domain with advanced security enabled:
+
+```hcl reference
+https://github.com/camunda/camunda-tf-eks-module/blob/feature/opensearch-doc/examples/camunda-8.6-irsa/opensearch.tf
+```
+
+Once the IRSA configuration is complete, ensure you **record the IAM role name** (from the `iam_opensearch_role_name` configuration), it is required to annotate the Kubernetes service account in the next step.
+
+As the OpenSearch domain has advanced security enabled and [fine-grained access control](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/fgac.html), we will later use your provided master username (`advanced_security_master_user_name`) and password (`advanced_security_master_user_password`) to perform the initial setup of the security component, allowing the created IRSA role to access the domain.
+
+</TabItem>
+</Tabs>
+
+#### Step 2: Additional customization
+
+You can further customize the OpenSearch cluster setup using various input options. For a full list of available parameters, see the [OpenSearch module documentation](https://github.com/camunda/camunda-tf-eks-module/blob/2.6.0/modules/opensearch/README.md).
+
+### Outputs
+
+**Terraform** allows you to define outputs, which make it easier to retrieve important values generated during execution, such as database endpoints and other necessary configurations for Helm setup.
+
+Each module that you have previously setup contains at the end the definition of the outputs. You can adjust them to your needs.
+
+These outputs will allow you to easily reference the **cert-manager** ARN, **external-dns** ARN, and the endpoints for both **PostgreSQL** and **OpenSearch** in subsequent steps or scripts, streamlining your deployment process.
+
+### Execution
```

Comment:
does it make sense to move the `terraform init` here?

Since we did it initially but now we've added between 1-3 extra modules depending on usage and require another tf init -upgrade to continue.

Reply:
nono, if you follow the instructions 1:1.

Create `cluster.tf`
> 3. Initialize Terraform
> terraform init

We add the modules for EKS and optionally for Postgres + OpenSearch

A lot further down the road.

> Execution
> terraform plan -out cluster.plan

problem is that at this point one would first have to run `terraform init` again as the modules have not been initialized yet.